core/object/object.cpp

```diff
@@ -1473,10 +1473,14 @@ String Object::tr(const StringName &p_message, const StringName &p_context) cons
 	}
 
 	if (Engine::get_singleton()->is_editor_hint()) {
-		return TranslationServer::get_singleton()->tool_translate(p_message, p_context);
-	} else {
-		return TranslationServer::get_singleton()->translate(p_message, p_context);
+		String tr_msg = TranslationServer::get_singleton()->extractable_translate(p_message, p_context);
+		if (tr_msg.is_empty() || tr_msg == p_message) {
```

Comment:
This condition will cause lots of long string comparisons. Any way to avoid it?

Reply:
I think I can drop the string comparison, albeit the cases where the first condition wouldn't be triggered first would be small.

Reply:
Done.